phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java

```diff
@@ -100,55 +100,20 @@ private void updateCounters(IndexTool.IndexVerifyType verifyType,
 
     @Override
     protected void reduce(ImmutableBytesWritable arg0, Iterable<IntWritable> arg1,
-                          Reducer<ImmutableBytesWritable, IntWritable, NullWritable, NullWritable>.Context context)
-            throws IOException, InterruptedException
-
-    {
+            Reducer<ImmutableBytesWritable, IntWritable, NullWritable, NullWritable>.Context context)
+            throws IOException, InterruptedException {
         if (!calledOnce.compareAndSet(false, true)) {
             return;
         }
-        IndexTool.IndexVerifyType verifyType = getIndexVerifyType(context.getConfiguration());
-        if (verifyType != IndexTool.IndexVerifyType.NONE) {
-            updateCounters(verifyType, context);
-        }
-    }
-
-    @Override
-    protected void cleanup(Context context) throws IOException, InterruptedException{
         try {
-            IndexToolUtil.updateIndexState(context.getConfiguration(), PIndexState.ACTIVE);
-
-            updateTasksTable(context);
+            IndexTool.IndexVerifyType verifyType = getIndexVerifyType(configuration);
+            if (verifyType != IndexTool.IndexVerifyType.NONE) {
+                updateCounters(verifyType, context);
+            }
+            IndexToolUtil.updateIndexState(configuration, PIndexState.ACTIVE);
         } catch (SQLException e) {
             LOGGER.error(" Failed to update the status to Active");
```

Comment:
This log statement might not be correct if the getIndexVerifyType or updateCounters throw an exception